plugins/actsvg/include/detray/plugins/actsvg/surface_svg_converter.hpp

```diff
@@ -0,0 +1,133 @@
+#include <math.h>
+
+#include <type_traits>
+#include <vector>
+
+// Project include(s)
+#include "actsvg/meta.hpp"
+#include "actsvg/proto/surface.hpp"
+#include "detray/definitions/units.hpp"
+#include "detray/geometry/surface.hpp"
+
+namespace detray::actsvg_visualization {
+
+using point3 = std::array<actsvg::scalar, 3>;
+using point3_container = std::vector<point3>;
+using proto_surface = actsvg::proto::surface<point3_container>;
+
+template <typename matrix_t>
+inline std::array<detray::scalar, 3> rotation_matrix_to_euler_angles(
+    const matrix_t& mat) {
+    float a = std::sqrt(mat[0][0] * mat[0][0] + mat[1][0] * mat[1][0]);
+    // Checking if it is singular.
+    if (a < 1e-6)
+        return {std::atan2(-mat[1][2], mat[1][1]), std::atan2(-mat[2][0], a),
+                0};
+
+    return {std::atan2(mat[2][1], mat[2][2]), std::atan2(-mat[2][0], a),
+            std::atan2(mat[1][0], mat[0][0])};
+}
+
+template <typename transform_t>
+inline auto convert_transform(const transform_t& d_transform) {
+    auto translation = d_transform.translation();
+    auto euler_angles =
+        rotation_matrix_to_euler_angles<>(d_transform.rotation());
+
+    // TODO: skew and scale
+
+    auto ret = actsvg::style::transform();
+    constexpr auto rad_to_deg = 180.0 / 3.14;
```

Comment:
Let's put this into ```detfinitions/units.hpp``` along the lines of what is already there, e.g. the conversion to degree